src/annotator/config/index.js

```diff
@@ -204,29 +213,36 @@ export function getConfig(appContext = 'annotator', window_ = window) {
       urlFromLinkTag(window_, 'sidebar', 'html')
     );
 
+    // Set an optional default first, this may be overridden below
+    if (hasDefault) {
+      config[name] = configDef.defaultValue;
+    }
+
     // Only allow certain values in the browser extension context
     if (!configDef.allowInBrowserExt && isURLFromBrowserExtension) {
-      // If the value is not allowed here, then set to the default if provided, otherwise ignore
-      // the key:value pair
-      if (hasDefault) {
-        config[name] = configDef.defaultValue;
-      }
       return;
     }
 
     // Get the value from the configuration source
     const value = configDef.getValue(settings, name);
     if (value === undefined) {
-      // If there is no value (e.g. undefined), then set to the default if provided,
-      // otherwise ignore the config key:value pair
-      if (hasDefault) {
-        config[name] = configDef.defaultValue;
-      }
       return;
     }
 
-    // Finally, run the value through an optional coerce method
-    config[name] = configDef.coerce ? configDef.coerce(value) : value;
+    // If there is a validator, run that next
+    const validationResult = configDef.validate
+      ? configDef.validate(value)
+      : true;
+    if (validationResult === true) {
+      // Finally, run the value through an optional coerce method
+      config[name] = configDef.coerce ? configDef.coerce(value) : value;
+    } else {
+      // Report the error to the user
+      console.error(
```

Comment:
So now we have a means of informing uses when bunk values are passed. In these cases, either the config will be omitted from the result, or if a default is set.